articles/working-with-apis-in-typescript/index.md

```diff
@@ -0,0 +1,296 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/working-with-apis-in-typescript/
+title: Working with APIs in TypeScript
+description: In this article, we will take a look at how to harness the power of TypeScript to consume APIs.
+author: saiharsha-balasubramaniam
+date: 2021-04-14T00:00:00-17:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/working-with-apis-in-typescript/hero.png
+    alt: TypeScript API example image
+---
+
+TypeScript was introduced by Microsoft as a superset of JavaScript. It takes the versatility of JavaScript and improves it with type definitions. TypeScript is used with a lot of popular frameworks like React.js and React Native to build enterprise applications.
+
+<!--more-->
+
+In this article, we shall take a look at how to harness the power of TypeScript to create an API.
+
+### Prerequisites & Pre-Reading Material
+
+- A basic understanding of [HTTP Requests](https://www.section.io/engineering-education/http-requests-nodejs/) and [APIs](https://www.section.io/engineering-education/rest-api/).
+- Fundamental concepts of [Node.js](https://www.section.io/engineering-education/history-of-nodejs/).
+- [TypeScript basics](https://www.section.io/engineering-education/a-friendly-beginner-guide-to-typescript/).
+
+### Project Setup
+
+To get started, let us create a working directory for our project:
+
+```bash
+# Create a directory 'node-typescript'
+mkdir node-typescript
+
+# Choose the new directory as the current directory
+cd node-typescript
+```
+
+Let us initialize a Node.js project:
+
+```bash
+# The -y option initializes the project with default settings
+npm init -y
+```
+
+Express is a framework built on Node.js. It is used to build backend servers and APIs. To learn more about Express, visit their [official docs](https://expressjs.com/en/4x/api.html).
```

Comment:
`https://expressjs.com/en/4x/api.html` This link seems to be broken. It shows 404. Could you please add a working link?

Reply:
![image](https://user-images.githubusercontent.com/34317373/114755909-c95e4680-9d77-11eb-8189-3c2b907ecba4.png)

This website seems to be working. Could you please check again?

Reply:
My bad. I guess I didn't copy the entire URL. It works fine.

Reply:
No worries